text/0004-structured_data_io.md

```diff
@@ -12,33 +11,49 @@ we'd hope, or less complete than we'd expect from a new RFC.
 # Summary
 [summary]: #summary
 
-Structured I/O defines interfaces for reading & writing streams of _entries_,
-which are elements of a composite data structure such as an array or object.
-Structured reader & writer interfaces combine a byte stream, data format 
-and schema to create entry readers & writers that produce & consume entries
-instead of bytes. Structured I/O streams can be composed & connected to form
-the basis of rich data communication capable of spanning across formats.
+Structured I/O defines interfaces for reading & writing streams of parsed 
+ _entries_, which are elements of dynamic-yet-structred data such as JSON, CBOR,
```

Comment:
Type: structred -> structured

Also, by "dynamic" I believe you mean "arbitrary"? Or, next paragraph, you use "generic-yet-structured".